### PR TITLE
Cluster Details and Settings Part 1

### PR DESCRIPTION
**Pull request by nelsonkopliku**
This PR is the first of a series related to Cluster detail, settings, checks selection and execution.
Added a couple of CTAs in the cluster detail to:
- Adjust settings of a cluster
- Look at the latest execution result

In addition saving the selected checks does not trigger execution automatically, that will be on demand

image

image

Next:
- API+UI for connection settings
- On demand execution request
- UI improvements

I'd like to keep PRs not that big. Sorry about the incomplete work.

### CODE REVIEW

File: assets/js/components/ClusterDetails/ClusterDetails.jsx [hunk 1]
```diff
@@ -18,6 +23,12 @@ const siteDetailsConfig = {
       className: 'table-col-m',
       render: (content) => content?.join(', '),
     },
+    {
+      title: 'Virtual IP',
+      key: '',
+      className: 'table-col-m',
+      render: (_, _item) => 'Where to get this?',
```
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterDetails.jsx:30]: We need to add some logic to get this value, and most probably add the field in the HANA read model value object.
To get the value, we need to parse the list of cluster resources, find the one with `ocf::heartbeat:IPaddr2` type, and get the value. This resource is only available in one of the nodes.
Let's put a follow up task.

File: assets/js/components/ClusterDetails/ClusterSettings.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+import { Tab } from '@headlessui/react';
+import { ChecksSelection } from './ChecksSelection';
+
+export const ClusterSettings = () => {
+  const { clusterID } = useParams();
+  const navigate = useNavigate();
+
+  const clusters = useSelector((state) => state.clustersList.clusters);
+  const cluster = clusters.find((cluster) => cluster.id === clusterID);
```
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterSettings.jsx:20]: Could we use the `getCluster` selector function here?
https://github.com/trento-project/web/blob/main/assets/js/state/selectors/index.js#L7

File: assets/js/components/ClusterDetails/ClusterSettings.jsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+  let [categories] = useState({
+    'Checks Selection': <ChecksSelection />,
+    'Connection Settings': (
```
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterSettings.jsx:24]: Is this some current placeholder?

File: assets/js/components/ClusterDetails/ClusterSettings.jsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import { ChecksSelection } from './ChecksSelection';
+
+export const ClusterSettings = () => {
+  const { clusterID } = useParams();
+  const navigate = useNavigate();
+
+  const clusters = useSelector((state) => state.clustersList.clusters);
+  const cluster = clusters.find((cluster) => cluster.id === clusterID);
+
+  let [categories] = useState({
```
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ClusterSettings.jsx:22]: Why do we have these entries in an `useState` function?
They look simple constants to me

**Review by arbulu89 — APPROVED** [review]
Great work!
Some comments noted.
I have doubts what will happen with the checks selection view if there is some connection issue with the runner. Does the loading and error notification pages work fine?

File: assets/js/components/ClusterDetails/ClusterSettings.jsx (hunk 3 above)
Comment (nelsonkopliku) [reply assets/js/components/ClusterDetails/ClusterSettings.jsx:24]: yes

File: assets/js/components/ClusterDetails/ClusterSettings.jsx (hunk 4 above)
Comment (nelsonkopliku) [reply assets/js/components/ClusterDetails/ClusterSettings.jsx:22]: Copy pasted, going fast and making it work :smile: 

It's gonna change probably.

File: assets/js/components/ClusterDetails/ClusterSettings.jsx (hunk 2 above)
Comment (nelsonkopliku) [reply assets/js/components/ClusterDetails/ClusterSettings.jsx:20]: definitely!

File: assets/js/components/ClusterDetails/ClusterDetails.jsx (hunk 1 above)
Comment (nelsonkopliku) [reply assets/js/components/ClusterDetails/ClusterDetails.jsx:30]: Yup https://github.com/trento-project/web/issues/389

**Comment by nelsonkopliku** [comment]
Thanks Xabi, gonna address things step by step.

> I have doubts what will happen with the checks selection view if there is some connection issue with the runner. Does the loading and error notification pages work fine?

Something bad happens if the runner does not work, as we load several times the catalog from there.
I don't have another answer right now. 
Definitely needs testing.

File: assets/js/components/ClusterDetails/ClusterSettings.jsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useParams, useNavigate } from 'react-router-dom';
+
+import { EOS_ARROW_BACK } from 'eos-icons-react';
+import Button from '@components/Button';
+
+function classNames(...classes) {
+  return classes.filter(Boolean).join(' ');
+}
```
Comment (dottorblaster) [inline assets/js/components/ClusterDetails/ClusterSettings.jsx:10]: `import classNames from classnames` will do

**Review by fabriziosestito — APPROVED** [review]
Great job! we had a live review about this, looking forward to the next iteration

**Review by dottorblaster — CHANGES_REQUESTED** [review]

**Review by dottorblaster — APPROVED** [review]

**Pull request merged**